internal/services/api/post_test_credentials.go

```diff
@@ -37,26 +39,57 @@ func PostTestCredentialsHandlerFunc(log logging.Logger) http.HandlerFunc {
 			BadRequest(w, req, log, err)
 			return
 		}
-		acct := &accounts.Account{
-			ServerType: accounts.ServerTypeConnect,
-			AuthType:   accounts.AuthTypeAPIKey,
-			URL:        b.URL,
-			Insecure:   b.Insecure,
-			ApiKey:     b.ApiKey,
-		}
 
-		timeout := time.Duration(max(b.Timeout, 30) * 1e9)
-		client, err := clientFactory(acct, timeout, nil, log)
+		// create a list of URLs to attempt
+		possibleURLs, err := util.GetListOfPossibleURLs(b.URL)
 		if err != nil {
-			InternalError(w, req, log, err)
+			BadRequest(w, req, log, err)
 			return
 		}
 
-		user, err := client.TestAuthentication(log)
+		// walk the possible URL list backwards (it has the full path first)
+		var urlToBeTested string
+		var lastTestError error
+		var user *connect.User
+
+		for i := len(possibleURLs) - 1; i >= 0; i-- {
+			urlToBeTested = possibleURLs[i]
+
+			acct := &accounts.Account{
+				ServerType: accounts.ServerTypeConnect,
+				AuthType:   accounts.AuthTypeAPIKey,
+				URL:        urlToBeTested,
+				Insecure:   b.Insecure,
+				ApiKey:     b.ApiKey,
+			}
+
+			timeout := time.Duration(max(b.Timeout, 30) * 1e9)
+			client, err := clientFactory(acct, timeout, nil, log)
+			if err != nil {
+				InternalError(w, req, log, err)
```

Comment:
I know this is the previous `InternalError` usage moved into here, but it'd be nice to start shying away on returning 500 errors when possible

Reply:
agreed, but let's address it in a separate PR as we eliminate the others. I'd like to keep this release-blocking PR as targeted as possible.